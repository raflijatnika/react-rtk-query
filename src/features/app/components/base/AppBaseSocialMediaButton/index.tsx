// React
import { memo } from 'react'

// Components
import { AppBaseButton } from '@/features/app/components'
import { StyledWrapper } from './components'

// i18n
import { useTranslation } from 'react-i18next'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

// Assets
import GoogleIcon from '@/assets/images/icon/google.png'
import FacebookIcon from '@/assets/images/icon/facebook.png'

const AppBaseSocialMediaButton = memo(() => {
  // Hook
  const { t } = useTranslation()

  return (
    <StyledWrapper data-testid='wrapper'>
      {/* Google */}
      <AppBaseButton
        data-testid='google'
        color={APP_COLOR.DARK}
        backgroundColor={APP_COLOR_LIGHT.BACKGROUND}
        width={164}
        height={50}
        icon={<img src={GoogleIcon} width={'24px'} height={'24px'} />}
      >
        {t('app.social.google')}
      </AppBaseButton>

      {/* Facebook */}
      <AppBaseButton
        data-testid='facebook'
        color={APP_COLOR.DARK}
        backgroundColor={APP_COLOR_LIGHT.BACKGROUND}
        width={164}
        height={50}
        icon={<img src={FacebookIcon} width={'9px'} height={'18px'} />}
      >
        {t('app.social.facebook')}
      </AppBaseButton>
    </StyledWrapper>
  )
})

AppBaseSocialMediaButton.displayName = 'AppBaseSocialMediaButton'

export { AppBaseSocialMediaButton }
