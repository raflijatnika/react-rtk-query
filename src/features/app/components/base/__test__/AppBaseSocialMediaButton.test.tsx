// Test
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

// Components
import { AppBaseSocialMediaButton } from '../AppBaseSocialMediaButton'

it('AppBaseSocialMediaButton should render', () => {
  render(<AppBaseSocialMediaButton />)
})

it('AppBaseSocialMediaButton should have snapshot', () => {
  const tree = renderer.create(<AppBaseSocialMediaButton />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('AppBaseSocialMediaButton should have google button and have 2 children', () => {
  render(<AppBaseSocialMediaButton />)
  const element = screen.getByTestId('google')
  expect(element.childElementCount).toEqual(2)
})

it('AppBaseSocialMediaButton should have facebook button and have 2 children', () => {
  render(<AppBaseSocialMediaButton />)
  const element = screen.getByTestId('facebook')
  expect(element.childElementCount).toEqual(2)
})
