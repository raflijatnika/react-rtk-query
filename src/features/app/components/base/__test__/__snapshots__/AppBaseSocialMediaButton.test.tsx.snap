// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBaseSocialMediaButton should have snapshot 1`] = `
<div
  className="sc-kDDrLX hWkIGw"
  data-testid="wrapper"
>
  <button
    className="ant-btn ant-btn-default sc-gsnTZi iOAFUR"
    color="#06152B"
    data-testid="google"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <img
      height="24px"
      src="google.png"
      width="24px"
    />
    <span>
      app.social.google
    </span>
  </button>
  <button
    className="ant-btn ant-btn-default sc-gsnTZi iOAFUR"
    color="#06152B"
    data-testid="facebook"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <img
      height="18px"
      src="facebook.png"
      width="9px"
    />
    <span>
      app.social.facebook
    </span>
  </button>
</div>
`;
